src/benchmarklib/tpch/tpch_table_generator.cpp

```diff
@@ -125,9 +126,24 @@ std::unordered_map<std::string, BenchmarkTableInfo> TPCHTableGenerator::generate
   Assert(_scale_factor < 1.0f || std::round(_scale_factor) == _scale_factor,
          "Due to tpch_dbgen limitations, only scale factors less than one can have a fractional part.");
 
-  const auto cache_directory = std::string{"tpch_cached_tables/sf-"} + std::to_string(_scale_factor);  // NOLINT
-  if (_benchmark_config->cache_binary_tables && std::filesystem::is_directory(cache_directory)) {
-    return _load_binary_tables_from_path(cache_directory);
+  auto cache_directory = std::string{};
+
+  if (_benchmark_config->use_mmap) {
+    if (!std::filesystem::is_directory("tpch_cached_tables_storage_json")) {
```

Comment:
Not to be addressed: why use a flag `mmap` instead of simply passing a directory name that automatically states if we want to use persistency and which would avoid such hard-coded names?